src/block/block_ckpt.c

```diff
@@ -952,3 +952,12 @@ __wt_block_checkpoint_resolve(WT_SESSION_IMPL *session, WT_BLOCK *block, bool fa
 
     return (ret);
 }
+
```

Comment:
One quick question before a full review. Did we consider
```
#ifdef HAVE_UNITTEST
    #define WT_STATIC
#else
    #define WT_STATIC static
#end

WT_STATIC int __ckpt_add_blkmod_entry(...)
```
to avoid all the
```
#ifdef HAVE_UNITTEST

int
__ut_ckpt_add_blkmod_entry(
  WT_SESSION_IMPL *session, WT_BLOCK_MODS *blk_mod, wt_off_t offset, wt_off_t len)
{
    return __ckpt_add_blkmod_entry(session, blk_mod, offset, len);
}
#endif
```
boilerplate?

Reply:
yeah, that gets annoying because of the auto-generated stuff in `extern.h`. there's also linkage issues that i never quite solved because i timeboxed the investigation

Reply:
The question of how we unit test static functions is an interesting one, and we should discuss the issue at a Storage Engines team level. Because we might even agree to get rid of static functions.
Given we have static functions now that we want to test there are a number of ways of exposing them for unit testing, and Andrew has given a couple of options (including the one Will and I are using). One think that makes me nervous about the `WT_STATIC` idea is that it is no longer obvious to someone calling the function that it will be non-static in a unit test, and therefore they might write non-test code that works in when unit tested, but no longer builds without the `HAVE_UNITTEST` flag because they are calling a non-accessible static function.
In C++ you can use friend if necessary.